cfgov/v1/wagtail_hooks.py

```diff
@@ -19,6 +19,11 @@
 logger = logging.getLogger(__name__)
 
 
+@hooks.register('before_delete_page')
+def raise_delete_error(request, page):
+    raise Exception('Deletion via POST is disabled')
```

Comment:
That's true. From a pragmatic point of view, the advantage of not raising an unhandled exception here is just to avoid treating it like some kind of failure (triggers error email, shows up in stats as a 500 error).

In production that error message won't appear either -- you'll see [our custom 500 page](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/cfgov/urls.py#L397) which says "500 internal server error". From a user point of view it won't be obvious that the behavior is unsupported versus that some error occurred.

From a more pedantic point of view, 500 Internal Server Error [is supposed to mean](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html)

> The server encountered an unexpected condition which prevented it from fulfilling the request.

That doesn't really apply in this case: we know how to handle the request, we just want to choose to deny it. 403 Forbidden, on the other hand, is supposed to mean

> The server understood the request, but is refusing to fulfill it.

You're right that in both debug and production right now raising `PermissionDenied` will produce an ugly "403 Forbidden" page (this is true of any other views that do this). We could override this with a custom template (much like we have done with our 404 and 500 pages).

From an even _more_ pedantic point of view, it's good to avoid `raise Exception` using the base exception class as it makes it difficult for code higher up the stack to know what happened. From [PEP8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations):

> Design exception hierarchies based on the distinctions that code catching the exceptions is likely to need, rather than the locations where the exceptions are raised. Aim to answer the question "What went wrong?" programmatically, rather than only stating that "A problem occurred"